src/Core/Services/Implementations/SendService.cs

```diff
@@ -142,10 +144,11 @@ public async Task<bool> ValidateSendFile(Send send)
 
             var (valid, realSize) = await _sendFileStorageService.ValidateFileAsync(send, fileData.Id, fileData.Size, _fileSizeLeeway);
 
-            if (!valid)
+            if (!valid || realSize > MAX_FILE_SIZE)
             {
                 // File reported differs in size from that promised. Must be a rogue client. Delete Send
                 await DeleteSendAsync(send);
```

Comment:
I've been thinking about this.... should we delete the Send, OR, just delete the file from the Send... Do you need to have a file on a File Send (i've not tried it, but could you create a File send without actually uploading a file at all?)

Reply:
You could have a send with no file attached (and in fact this happens in transient while uploading).

If we instead only delete the file, what will happen on the Access and add-edit pages is this:
All metadata will be correct (file size, file name, other we add later), but the endpoint for download will not be found. clicking the download link will resul in an error clients should display to that effect as well, but it won't be clear _why_ no file is found.

We could have a usually-hidden indicator that informed the user that we detected abuse and so delete the file,but not the send. However, this should code path should only be triggered when someone does some pretty fishy stuff. That seems like more trouble than its worth to explain to someone abusing our service where their data is.

Reply:
As far as creating a send without a file by "supported" channels. No. I think all our clients would probably balk at the resulting Send model as well. We just assume some things to not be `null` in lots of places.

Reply:
Makes perfect sense, thanks for the explanation and walkthrough.